### DISCUSSION TITLE
Proposal: Introduce prefix to use string interpolation with invariant culture

### DISCUSSION THREAD

**Discussion by drieseng** [dotnet/csharplang#7223] [Language Ideas]
Right now, using string interpolation with invariant culture is quite verbose:

**Invariant culture**:
```cs
using System.Globalization;
var x = string.Create(CultureInfo.InvariantCulture, $".....");
```

**Current culture**:
```cs
var x = $".....";
```

### Proposal

Support an "i" prefix to indicate that the invariant culture should be used for formatting:

**Invariant culture**:
```cs
var x = $i".....";
```

**Comment by CyrusNajmabadi** [comment]
Could just make an extension method of `.I()` in this case.  So `$"...".I()`.

**Comment by Rekkonnect** [comment]
Getting a language feature specifically for invariant culture string interpolation seems to be too specific a case. If we broaden the scope to this kind of features, the language would have to reserve all kinds of letters and symbols for every imaginable shortcut for existing methods.

This more looks like a byproduct of the lack of macros, which is another proposal on its own and has very mixed opinions, including the negative opinion from the LDT due to the sheer complexity of implementing macros.

**Reply by drieseng** [reply]
@Rekkonnect, then what's your view on UTF-8 string literals and #7072? Do you think that's more common ("less specific") than culture invariant formatting?

**Reply by Rekkonnect** [reply]
UTF-8 strings offer considerable performance improvements in specific scenarios for low-level code. This one definitely deserves a native way to express UTF-8 string literals.

Invariant culture string interpolation is not something that would give performance improvement, if it is to offer any at all. Plus, it's such an uncommon operation, most of the time string interpolation without bothering about the culture itself suffices. Given its rarity and insignificance other than an alternative way to express the same thing, this is really not worth any effort designing the syntax itself, let alone implementing it and learning it as a user of the language.

**Reply by drieseng** [reply]
@Rekkonnect, why would you think that the use of invariant culture is uncommon? How would you perform formatting in a way that is not specific to the system (current culture) your application is running on?

Perhaps it's not [that](https://grep.app/search?q=string.Format%28CultureInfo.InvariantCulture&case=true) common for OSS projects, but for backends in LOB applications it definitely is.

I'm not against UTF-8 strings, but for backends it is probably much less common than the use of invariant culture.

**Comment by iam3yal** [comment]
duplicate of #177
